File: scripts/update_projects_software.js
const fs = require('fs');
const path = require('path');

const filePath = path.join('c:', 'Users', 'USER', 'Documents', 'portfolio-shared', 'src', 'data', 'projects.json');

try {
    const fileContent = fs.readFileSync(filePath, 'utf8');
    const data = JSON.parse(fileContent);

    data.projects = data.projects.map(project => {
        if (project.slug === 'kampanye-hadiah-digital-liburan') {
            return { ...project, software: ['canva', 'photoshop'] };
        } else {
            return { ...project, software: ['photoshop'] };
        }
    });

    fs.writeFileSync(filePath, JSON.stringify(data, null, 2), 'utf8');
    console.log("Successfully updated projects.json with software metadata using Node.js.");
} catch (error) {
    console.error("Error updating projects.json:", error);
    process.exit(1);
}

File: scripts/update_projects_software.py
import json

file_path = r'c:\Users\USER\Documents\portfolio-shared\src\data\projects.json'

with open(file_path, 'r', encoding='utf-8') as f:
    data = json.load(f)

for project in data['projects']:
    if project['slug'] == 'kampanye-hadiah-digital-liburan':
        project['software'] = ['canva', 'photoshop']
    else:
        project['software'] = ['photoshop']

with open(file_path, 'w', encoding='utf-8') as f:
    json.dump(data, f, indent=2, ensure_ascii=False)

print("Successfully updated projects.json with software metadata.")
